randovania/generator/dock_weakness_distributor.py

```diff
@@ -279,7 +279,9 @@ def _determine_valid_weaknesses(
 
     if state is not None:
         reach = ResolverReach.calculate_reach(logic, state)
-        if state.node == target:
+        if state.node == target or isinstance(state.node, DockLockType) and state.node.dock == target:
```

Comment:
```suggestion
        if state.node == target or (isinstance(state.node, DockLockType) and state.node.dock == target):
```

Reply:
never trust operator precedence in conditionals, lmao